generator/transit_generator.hpp

```diff
@@ -1,9 +1,90 @@
+#pragma once
+
+#include "geometry/point2d.hpp"
+
+#include "base/macros.hpp"
+
+#include "3party/jansson/myjansson.hpp"
+
+#include <cstdint>
 #include <string>
+#include <type_traits>
+#include <vector>
 
 namespace routing
 {
 namespace transit
 {
-void BuildTransit(std::string const & mwmPath, std::string const & srtmDir);
+class DeserializerFromJson
+{
+public:
+  DeserializerFromJson(json_struct_t * node) : m_node(node) {}
+
+  template<typename T>
+  typename std::enable_if<std::is_integral<T>::value || std::is_enum<T>::value>::type
+      operator()(T & t, char const * name = nullptr)
+  {
+    GetField(t, name);
+  }
+
+  template<typename T>
+  typename std::enable_if<std::is_floating_point<T>::value>::type operator()(T & t, char const * name = nullptr)
+  {
+    NOTIMPLEMENTED();
+  }
+
+  void operator()(std::string & s, char const * name = nullptr) { GetField(s, name); }
+  void operator()(m2::PointD & p, char const * name = nullptr);
+
+  template <typename T>
+  void operator()(std::vector<T> & vs, char const * name = nullptr)
+  {
+    auto * arr = my::GetJSONOptionalField(m_node, name);
+    if (arr == nullptr)
+      return;
+
+    if (!json_is_array(arr))
+      MYTHROW(my::Json::Exception, ("The field", name, "must contain a json array."));
+    size_t const sz = json_array_size(arr);
+    vs.resize(sz);
+    for (size_t i = 0; i < sz; ++i)
+    {
+      DeserializerFromJson arrayItem(json_array_get(arr, i));
+      arrayItem(vs[i]);
+    }
+  }
+
+  template<typename T>
+  typename std::enable_if<std::is_class<T>::value>::type operator()(T & t, char const * name = nullptr)
+  {
+    t.Visit(*this);
+  }
+
+private:
+  template <typename T>
+  void GetField(T & t, char const * name = nullptr)
+  {
+    if (name == nullptr)
+    {
+      // |name| is not set in case of array items
+      FromJSON(m_node, t);
```

Comment:
Please, place a return here (you are returning early in the other branch anyway).

Reply:
Done

Reply:
> What about returning a bool from this method?

@mpimenov I answer the question above in details. In short I can do that but I don't see how I'll use the return value.